mountpoint-s3-client/src/s3_crt_client.rs

```diff
@@ -93,6 +93,7 @@ pub struct S3ClientConfig {
     max_attempts: Option<NonZeroUsize>,
     read_backpressure: bool,
     initial_read_window: usize,
+    network_interface_names: Option<Vec<String>>,
```

Comment:
Is there a difference between `None` and `Some(vec![])`? I think probably not, so this might not need to be Option.